lib/Importer.php

```diff
@@ -211,7 +211,8 @@ private function _addCertificateMetaData(&$parsedMetaData)
 
         foreach ($certKeys as $certKey) {
             if (isset($parsedMetaData[$certKey . 'X509Certificate']) &&
-                (!isset($parsedMetaData[$certKey . 'encryption']) ||
+                ( (isset($parsedMetaData[$certKey . 'signing']) && $parsedMetaData[$certKey . 'signing']) ||
+                    !isset($parsedMetaData[$certKey . 'encryption']) ||
```

Comment:
I can indeed factor out the first condition like that, see extra commit. However, the other three are or'ed so not easily split out into individual "continued" checks.